### DISCUSSION TITLE
Docker.img filling and I don't know why

### DISCUSSION THREAD

**Discussion by addohm** [abraunegg/onedrive#2599] [Q&A] [answered]
Im using unraid.  There is a docker app, which essentially just creates a docker container and assigns four variables to it.  All the files are kept physically kept on the machine.  For some reason, one of my onedrive instances is blown up to nearly the max capicity of my docker image, and the other onedrive instance is ramping up in size.
<img width="697" alt="image" src="https://github.com/abraunegg/onedrive/assets/25333107/d5386971-b9a5-4860-bba1-21618fd005aa">
BOTH docker image, as stated, store config and data files on the host, not in the container...
<img width="1126" alt="image" src="https://github.com/abraunegg/onedrive/assets/25333107/e6f102e7-b7ec-4e81-b2dc-6536d3859d96">
That said, I have _no_ idea how the size is almost 40G, but I would imagine this is the source of the problems I've had twice in the past where docker just stops working all together and I have to redo everything.  What would cause this?  How can I investigate and prevent this?  Can I fix it in its current state, and how?

Edit: Sorry that this is in Q&A, I didn't realize it.

**Comment by abraunegg** [comment, marked as answer]
@addohm 
> Docker.img filling and I don't know why

This will be because your 'unraid' platform Docker configuration is incorrect.

You need to ensure you configure the 'ONEDRIVE_DATA_DIR' variable to point at your large storage volume outside of the Docker environment. 

Please read https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md in greater detail and then work with 'unraid' support to fix your issue.

**Reply by addohm** [reply]
Will do.  I suppose I could go back to my old docker compose files before I switched to unraid.  Wouldn't take much.

**Reply by addohm** [reply]
<img width="196" alt="image" src="https://github.com/abraunegg/onedrive/assets/25333107/fbdf04cb-94ad-4de6-ad0d-8d0d16bac6c3">
>.<  Looks like you're bang on.

**Reply by addohm** [reply]
Im a little unclear about $ONEDRIVE_DATA_DIR.  It looks like it is defined on the host machine, and is not actually a docker container variable.  With the current onedrive container, the data and config volumes are defined and attached to the container using the appropriate paths defined in the documentation.  Unless theres some doings in the background looking for that variable in the container, I'm not sure this is what's wrong.

**Comment by abraunegg** [comment]
@addohm 
The docker documentation provided here is correct and works without issue. This has been tested and validated many times.

You are having issues with 'unraid' who are providing your Docker version. Please seek support from 'unraid' as to why their platform and plugin is not working.

**Reply by addohm** [reply]
Im not doubting the documention my friend.  Not at all!  In fact, when my server was debian, I'd ran this many times using your very extensive documentation :)  Im just simply saying that the container mounts exist.  Whether that host environment variable exists or not being a necessity is my question. 
<img width="239" alt="image" src="https://github.com/abraunegg/onedrive/assets/25333107/c0a056e4-a8ed-4be0-8c6f-9e92759df5ce">
 Ill pull out my old compose files and see how that fits here.  if it works I just need to figure out how to make a proper app template for the unraid community to use.

**Reply by addohm** [reply]
Yeah, even my old compose files didnt use that host env var.  
<img width="532" alt="image" src="https://github.com/abraunegg/onedrive/assets/25333107/5225206a-bb98-4710-a426-3df2073c84dd">

**Reply by addohm** [reply]
How about this question - what is filling up the image file?  The data is definetly getting stored host, and the config is...well its the config but also on the physical host.  The log files are miniscule, so what's left?  Im more than happy to go to unraid but its literally just a layer that interfaces between the UI and the docker compose and theyre just going to tell me to come back here..

**Reply by abraunegg** [reply]
Are you sure the files are not being stored inside the docker container?

Did you enable logging?

When you login to the console of the Docker image itself - poke around the file system and see what is taking up the disk space.

No one else is reporting this issue ... so it is either 'unraid' or your configuration.

Given your network issues with Docker on 'unraid' ... I would be looking there.

**Comment by addohm** [comment]
Somehow, at some point, onedrive was storing files in `/root/OneDrive/` instead of the defined and mapped data folder.
<img width="163" alt="image" src="https://github.com/abraunegg/onedrive/assets/25333107/7a2221a4-de09-466d-91de-b5557fa369b0">

**Reply by addohm** [reply]
We tried to muscle it and just change the mapping to `/root/OneDrive | .config` but then it just went ahead and used `/onedrive/data | config` anyway, so no joy.  The question really becomes why did it ever put anything in `/root`.  As I understand it, its not a default anywhere whether its dockerized or not.

**Comment by abraunegg** [comment]
@addohm 
>Somehow, at some point, onedrive was storing files in /root/OneDrive/ instead of the defined and mapped data folder.

This means one of 3 things:
* You are running Docker as 'root'
* You are running the 'onedrive' Docker container as 'root'
* 'unraid' is running the Docker container as 'root'

Running Docker as root poses security risks. If an attacker gains control over the Docker daemon, they effectively have root access to the host system. This is a significant concern in terms of system security.

To mitigate this risk, Docker introduced the concept of rootless mode. Running Docker in rootless mode means the Docker daemon and containers run as an unprivileged user. This increases security as it restricts the scope of what can be done within the Docker environment, reducing the risk of a compromised Docker container affecting the host system.

The [Docker 'onedrive' documentation](https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md) specifically advises to configure your Docker platform to not run as root.

So *again* ... this is a 'unraid' problem ... please seek support from 'unraid'
